components/idp-mgt/org.wso2.carbon.idp.mgt/src/test/java/org/wso2/carbon/idp/mgt/dao/CacheBackedIdPMgtDAOTest.java

```diff
@@ -0,0 +1,1486 @@
+                {SAMPLE_TENANT_ID, TENANT_DOMAIN, "tes_I*", 2},
+                {SAMPLE_TENANT_ID, TENANT_DOMAIN, "*1", 1},
+                {SAMPLE_TENANT_ID, TENANT_DOMAIN, "Notexist", 0},
+        };
+    }
+
+    @Test(dataProvider = "getIdPsSearchData")
+    public void testGetIdPsSearch(int tenantId, String tenantDomain, String filter, int resultCount) throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDataSource()).thenReturn(dataSourceMap.get(DB_NAME));
+            addTestIdps();
+
+            // Directly providing initiated db connection as connection parameter
+            List<IdentityProvider> idpList1 = cacheBackedIdPMgtDAO.getIdPsSearch(connection, tenantId,
+                    tenantDomain, filter);
+            assertEquals(idpList1.size(), resultCount);
+            // Providing null as connection parameter so value of connection parameter will be retrieved
+            // from the mock above
+            List<IdentityProvider> idpList2 = cacheBackedIdPMgtDAO.getIdPsSearch(null, tenantId,
+                    tenantDomain, filter);
+            assertEquals(idpList2.size(), resultCount);
+        }
+    }
+
+    @Test(dataProvider = "getIdPsSearchData")
+    public void testGetIdPsSearchException(int tenantId, String tenantDomain, String filter, int resultCount)
+            throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDataSource()).thenReturn(dataSourceMap.get(DB_NAME));
+            addTestIdps();
+        }
+        assertThrows(IdentityProviderManagementException.class, () ->
+                cacheBackedIdPMgtDAO.getIdPsSearch(null, tenantId, tenantDomain, filter));
+    }
+
+    @DataProvider
+    public Object[][] getPaginatedIdPsSearchData() {
+
+        ExpressionNode expressionNode = new ExpressionNode();
+        List<ExpressionNode> expressionNodeList = new ArrayList<>();
+        expressionNodeList.add(expressionNode);
+
+        List<String> attributesList1 = Arrays.asList("id", "name", "description", "isEnabled", "image", "isPrimary");
+        List<String> attributesList2 = Arrays.asList("homeRealmIdentifier", "isFederationHub", "certificate", "alias",
+                "claims", "roles", "federatedAuthenticators", "provisioning");
+
+        return new Object[][]{
+                {SAMPLE_TENANT_ID, expressionNodeList, 2, 0, "ASC", "NAME", attributesList1, 2},
```

Comment:
Same value passed for all combination as SAMPLE_TENANT_ID

Reply:
Resolved with https://github.com/wso2/carbon-identity-framework/pull/4110/commits/28a1194ed9f5de659bd1879f625cd070efdac3fa